// dvd-service-js-util/dvd-service-js-util.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _dvdBaseJsUa = require('dvd-base-js-ua');

var _dvdBaseJsUa2 = _interopRequireDefault(_dvdBaseJsUa);

var _dvdServiceJsNative = require('dvd-service-js-native');

var _dvdServiceJsNative2 = _interopRequireDefault(_dvdServiceJsNative);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * @module dvd-service-js-util
 * @author swg [源码地址](http://gitlab.rd.vyohui.com/FE-Service/dvd-service-js-util.git)
 */
exports.default = {
  /**
   * 获取文档滚动条-上方距离
   */
  getDocumentScrollTop: function getDocumentScrollTop() {
    return document.body.scrollTop || window.pageYOffset || document.documentElement.scrollTop || 0;
  },

  /**
   * 获取文档滚动条-左侧距离
   */
  getDocumentScrollLeft: function getDocumentScrollLeft() {
    return document.body.scrollLeft || window.pageXOffset || document.documentElement.scrollLeft || 0;
  },

  /**
   * 页面触底通知回调
   */
  pageScrollToBottom: function pageScrollToBottom(callback, callback_ts) {
    var ts = this;

    var lastScrollTop = 0;
    var currentScrollTop = 0;
    window.addEventListener('scroll', function () {
      currentScrollTop = ts.getDocumentScrollTop();
      if (currentScrollTop > lastScrollTop && document.documentElement.clientHeight + currentScrollTop >= document.body.clientHeight * 0.9) {
        callback_ts ? callback.call(callback_ts) : callback();
      }
      lastScrollTop = currentScrollTop;
    }, false);

    var lastY = 0;
    var currentY = 0;
    window.addEventListener('touchmove', function (event) {
      currentY = event.changedTouches[0].clientY;
      if (currentY < lastY && document.documentElement.clientHeight + currentScrollTop >= document.body.clientHeight * 0.9) {
        callback_ts ? callback.call(callback_ts) : callback();
      }
      lastY = currentY;
    }, false);
  },
  getOffsetLeft: function getOffsetLeft(node) {
    if (node.offsetParent) {
      return node.offsetLeft + this.getOffsetLeft(node.offsetParent);
    } else {
      return node.offsetLeft;
    }
  },
  getOffsetTop: function getOffsetTop(node) {
    if (node.offsetParent) {
      return node.offsetTop + this.getOffsetTop(node.offsetParent);
    } else {
      return node.offsetTop;
    }
  },

  /**
   * 获取页面基础域名
   */
  getBaseDomain: function getBaseDomain() {
    return (/(bravetime.net|vyohui.cn|davdian.com)/.exec(location.host)[0]
    );
  },

  /**
   * 获取页面二级域名
   */
  getSecondDomain: function getSecondDomain() {
    return (/(.*)[.](bravetime.net|vyohui.cn|davdian.com)/.exec(location.host)[1]
    );
  },

  /**
   * 调用该方法，说明页面需要返回刷新
   */
  backReload: function backReload() {
    var addListener = function addListener() {
      // 延迟添加事件，防止页面加载完毕后立即触发
      setTimeout(function () {
        window.addEventListener('pageshow', function () {
          location.reload();
        }, false);
      }, 1);
    };
    // 页面加载完毕后添加pageshow事件监听，只在返回时触发
    if (document.readyState === 'complete') {
      addListener();
    } else {
      window.addEventListener('load', addListener, false);
    }
  },

  /**
   * 打开新页面
   */
  open: function open(url) {
    if (_dvdBaseJsUa2.default.isDvdApp()) {
      _dvdServiceJsNative2.default.Browser.open({
        url: url
      });
    } else if (_dvdBaseJsUa2.default.isWeiXin()) {
      // 微信不支持打开新页面，且ios微信甚至连window.open方法都不支持
      location.href = url;
    } else {
      window.open(url);
    }
  }
};
